scripts/pull_image_descriptions.py

```diff
@@ -0,0 +1,542 @@
+#!/usr/bin/env python3
+
+import argparse
+import fnmatch
+import json
+import pathlib
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+from datetime import datetime
+from typing import Dict, List, Tuple
+
+CONTAINER_IMAGE = "ghcr.io/osbuild/image-builder-cli:latest"
+GENERATION_DATE = datetime.now().strftime("%Y-%m-%d %H:%M:%S UTC")
+TARGET_DIR = pathlib.Path(__file__).parent.parent / "docs" / "user-guide" / "09-image-descriptions"
+
+
+def run_command(cmd: List[str]) -> Tuple[bool, str, str]:
+    """
+    Run a command and return success status, stdout, and stderr.
+    """
+    try:
+        result = subprocess.run(cmd, capture_output=True, text=True, check=True)
+        return True, result.stdout, result.stderr
+    except subprocess.CalledProcessError as e:
+        return False, e.stdout, e.stderr
+
+
+def pull_container_image() -> bool:
+    """
+    Pull the latest image-builder-cli container image.
+    """
+    print("Pulling latest container image...")
+    pull_cmd = ["sudo", "podman", "pull", CONTAINER_IMAGE]
+    success, _, stderr = run_command(pull_cmd)
```

Comment:
(super-nitpick) I personally would just write this via plain subprocess.run() without the helper, its the same mount of lines, the "check=..." is more repetitive but the upside is that when reading this as a newcomer its very clear what is happening without having to go and look into run_command(). I.e. just
```python3
res = subprocess.run(pull_cmd, check=False, capture_output=True, text=True)
if res.returncode != 0:
   print(f"Warning: Failed to pull container image: {res.stderr}")
   return False
print("Container image pulled successfully")
return True
```
but fine of course if you prefer it this way (and its already merged)